src/post/content.ts

```diff
@@ -207,6 +212,44 @@ export class ContentPreparer {
         return new Set(mentions);
     }
 
+    /**
+     * Updates existing hyperlinks with data-profile attribute and rel attributes
+     */
+    private updateMentions(content: string, mentions: Mention[]) {
+        try {
+            const html = parse(content);
+            const links = html.querySelectorAll('a');
+
+            for (const mention of mentions) {
+                const apId = mention.account.apId
+                    .toString()
+                    .replace(/\/+$/, '');
+                const url = mention.account.url.toString().replace(/\/+$/, '');
```

Comment:
I would love to see this abstracted a little bit so this `updateMentions` is focused more on "what" than "how"

Maybe a utility like:
```ts
equals(a: URL | string, b: URL | string) {
  if (a instanceof URL) return equals(a.href, b);
  if (b instanceof URL) return equals(a, b.href);
  
  return a.replace(trailingSlashRegex, '') === b.replace(trailingSlashRegex, '');
}
```

```ts
for (mention of mentions) {
  for (link of links) {
    if (equals(mention.account.apId, link.href) || equals(mention.account.url, link.href)) {
      // update da link
    }
  }
}
```

Reply:
Added here ; https://github.com/TryGhost/ActivityPub/pull/716